File: KnowledgeSpaceApk/adminapp/src/main/java/com/example/adminapp/MyPosts/MyPostActAdm.java
package com.example.adminapp.MyPosts;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.view.View;
import com.example.adminapp.databinding.ActivityMyPostActAdmBinding;

import java.util.ArrayList;

public class MyPostActAdm extends AppCompatActivity {
    private ActivityMyPostActAdmBinding binding;
    private adapterMyPostsAdm adapter;
    private ArrayList<dataModelRecVMyPostsAdm> dataHolder;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMyPostActAdmBinding.inflate(getLayoutInflater());
        View view = binding.getRoot();
        setContentView(view);

        binding.recyclerVMyPostActAdm.setLayoutManager(new LinearLayoutManager(getApplicationContext()));
        getData();
        adapter = new adapterMyPostsAdm(dataHolder);
        binding.recyclerVMyPostActAdm.setAdapter(adapter);

    }

    private void getData() {
        dataHolder = new ArrayList<>();
        //Add data
        dataModelRecVMyPostsAdm obj1 = new dataModelRecVMyPostsAdm("Demo Text Here With A Title Demoooooooooooo",
                "Demo desakpasdmpsampdmpll kwdmakmwpam ejjpjfs fkae pojpwmpd apo pmdp wm " +
                        "knpiwpadmpowadpm owapdmpwmadpj odwamwmdwapdw omadpkdwmdp");dataHolder.add(obj1);
        dataModelRecVMyPostsAdm obj2 = new dataModelRecVMyPostsAdm("Demo Text Here With A Title Demoooooooooooo",
                "Demo desakpasdmpsampdmpll kwdmakmwpam ejjpjfs fkae pojpwmpd apo pmdp wm " +
                        "knpiwpadmpowadpm owapdmpwmadpj odwamwmdwapdw omadpkdwmdp");dataHolder.add(obj2);
        dataModelRecVMyPostsAdm obj3 = new dataModelRecVMyPostsAdm("Demo Text Here With A Title Demoooooooooooo",
                "Demo desakpasdmpsampdmpll kwdmakmwpam ejjpjfs fkae pojpwmpd apo pmdp wm " +
                        "knpiwpadmpowadpm owapdmpwmadpj odwamwmdwapdw omadpkdwmdp");dataHolder.add(obj3);
        dataModelRecVMyPostsAdm obj4 = new dataModelRecVMyPostsAdm("Demo Text Here With A Title Demoooooooooooo",
                "Demo desakpasdmpsampdmpll kwdmakmwpam ejjpjfs fkae pojpwmpd apo pmdp wm " +
                        "knpiwpadmpowadpm owapdmpwmadpj odwamwmdwapdw omadpkdwmdp joaijspdjpwajdpw pjpwadmpmwpwmpo pjsapdjpwmpwajwm pojdpampwmp");dataHolder.add(obj4);
        dataModelRecVMyPostsAdm obj5 = new dataModelRecVMyPostsAdm("Demo Text Here With A Title Demoooooooooooo",
                "Demo desakpasdmpsampdmpll kwdmakmwpam ejjpjfs fkae pojpwmpd apo pmdp wm " +
                        "knpiwpadmpowadpm owapdmpwmadpj odwamwmdwapdw omadpkdwmdp");dataHolder.add(obj5);
        dataModelRecVMyPostsAdm obj6 = new dataModelRecVMyPostsAdm("Demo Text Here With A Title Demoooooooooooo",
                "Demo desakpasdmpsampdmpll kwdmakmwpam ejjpjfs fkae pojpwmpd apo pmdp wm " +
                        "knpiwpadmpowadpm owapdmpwmadpj odwamwmdwapdw omadpkdwmdp");dataHolder.add(obj6);
        dataHolder.add(obj5);dataHolder.add(obj5);dataHolder.add(obj5);dataHolder.add(obj5);dataHolder.add(obj5);



    }
}

File: KnowledgeSpaceApk/adminapp/src/main/java/com/example/adminapp/MyPosts/MyPostActAdm.kt
package com.example.adminapp.MyPosts

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.adminapp.databinding.ActivityMyPostActAdmBinding

class MyPostActAdm : AppCompatActivity() {
    private var binding: ActivityMyPostActAdmBinding? = null
    private var adapter: adapterMyPostsAdm? = null
    private var dataHolder: ArrayList<dataModelRecVMyPostsAdm>? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMyPostActAdmBinding.inflate(
            layoutInflater
        )
        val view: View = binding!!.root
        setContentView(view)
        binding!!.recyclerVMyPostActAdm.layoutManager = LinearLayoutManager(applicationContext)
        data
        adapter = adapterMyPostsAdm(dataHolder)
        binding!!.recyclerVMyPostActAdm.adapter = adapter
    }

    private val data: Unit
        private get() {
            dataHolder = ArrayList()
            //Add data
            val obj1 = dataModelRecVMyPostsAdm(
                "Demo Text Here With A Title Demoooooooooooo",
                "Demo desakpasdmpsampdmpll kwdmakmwpam ejjpjfs fkae pojpwmpd apo pmdp wm " +
                        "knpiwpadmpowadpm owapdmpwmadpj odwamwmdwapdw omadpkdwmdp"
            )
            dataHolder!!.add(obj1)
            val obj2 = dataModelRecVMyPostsAdm(
                "Demo Text Here With A Title Demoooooooooooo",
                "Demo desakpasdmpsampdmpll kwdmakmwpam ejjpjfs fkae pojpwmpd apo pmdp wm " +
                        "knpiwpadmpowadpm owapdmpwmadpj odwamwmdwapdw omadpkdwmdp"
            )
            dataHolder!!.add(obj2)
            val obj3 = dataModelRecVMyPostsAdm(
                "Demo Text Here With A Title Demoooooooooooo",
                "Demo desakpasdmpsampdmpll kwdmakmwpam ejjpjfs fkae pojpwmpd apo pmdp wm " +
                        "knpiwpadmpowadpm owapdmpwmadpj odwamwmdwapdw omadpkdwmdp"
            )
            dataHolder!!.add(obj3)
            val obj4 = dataModelRecVMyPostsAdm(
                "Demo Text Here With A Title Demoooooooooooo",
                "Demo desakpasdmpsampdmpll kwdmakmwpam ejjpjfs fkae pojpwmpd apo pmdp wm " +
                        "knpiwpadmpowadpm owapdmpwmadpj odwamwmdwapdw omadpkdwmdp joaijspdjpwajdpw pjpwadmpmwpwmpo pjsapdjpwmpwajwm pojdpampwmp"
            )
            dataHolder!!.add(obj4)
            val obj5 = dataModelRecVMyPostsAdm(
                "Demo Text Here With A Title Demoooooooooooo",
                "Demo desakpasdmpsampdmpll kwdmakmwpam ejjpjfs fkae pojpwmpd apo pmdp wm " +
                        "knpiwpadmpowadpm owapdmpwmadpj odwamwmdwapdw omadpkdwmdp"
            )
            dataHolder!!.add(obj5)
            val obj6 = dataModelRecVMyPostsAdm(
                "Demo Text Here With A Title Demoooooooooooo",
                "Demo desakpasdmpsampdmpll kwdmakmwpam ejjpjfs fkae pojpwmpd apo pmdp wm " +
                        "knpiwpadmpowadpm owapdmpwmadpj odwamwmdwapdw omadpkdwmdp"
            )
            dataHolder!!.add(obj6)
            dataHolder!!.add(obj5)
            dataHolder!!.add(obj5)
            dataHolder!!.add(obj5)
            dataHolder!!.add(obj5)
            dataHolder!!.add(obj5)
        }
}